### PR TITLE
Tests: give seed on default tests and fix execnet error

### PR DESCRIPTION
**Pull request by Silvris**
## What is this fixing or adding?
https://github.com/ArchipelagoMW/Archipelago/actions/runs/9488885076/job/26148887892

A test failure that gives effectively no information about the failure. ~~With these changes, we catch the exception being thrown such that the seed passed into `subTest` will get printed into the logs, allowing for reproduction of the error locally. While this is a stopgap solution, I couldn't find much information about the cause of the error itself.~~

The issue is that we pass `location` into `self.subTest`, when really all we want there is `location.name`. This PR instead fixes this issue alongside providing seed on failing tests.

## How was this tested?
~~https://github.com/Silvris/Archipelago/actions/runs/9489978465/job/26152426677~~
https://github.com/Silvris/Archipelago/actions/runs/9533282014/job/26276353017
Manually, by forcefully failing the test within the for loop of all_state_can_reach_everything.

Unittests were also ran locally.

### CODE REVIEW

File: test/bases.py [hunk 1]
```diff
@@ -292,23 +292,26 @@ def test_all_state_can_reach_everything(self):
...
-            for location in self.multiworld.get_locations():
-                if location.name not in excluded:
-                    with self.subTest("Location should be reached", location=location):
-                        reachable = location.can_reach(state)
-                        self.assertTrue(reachable, f"{location.name} unreachable")
-            with self.subTest("Beatable"):
-                self.multiworld.state = state
-                self.assertBeatable(True)
+        with self.subTest("Game", game=self.game, seed=self.multiworld.seed):
+                excluded = self.multiworld.worlds[self.player].options.exclude_locations.value
```
Comment (NewSoupVi) [inline test/bases.py:296]: Should this actually be indented an extra time now? If so, why?

Comment (Silvris) [reply test/bases.py]: (it's called I got lazy with removing an earlier try/except).

File: test/bases.py [hunk 2]
```diff
@@ -292,12 +292,12 @@ def test_all_state_can_reach_everything(self):
...
         if not (self.run_default_tests and self.constructed):
             return
-        with self.subTest("Game", game=self.game):
+        with self.subTest("Game", game=self.game, seed=self.multiworld.seed):
             excluded = self.multiworld.worlds[self.player].options.exclude_locations.value
             state = self.multiworld.get_all_state(False)
             for location in self.multiworld.get_locations():
                 if location.name not in excluded:
-                    with self.subTest("Location should be reached", location=location):
+                    with self.subTest("Location should be reached", location=location.name):
```
Comment (alwaysintreble) [inline test/bases.py:300]: does this keep failing the serialization dumping? i thought we had a proper repr for location, but if we don't i think that's probably the better fix for this

Comment (Silvris) [reply test/bases.py:300]: This is the cause of the serialization function failing within tests.

Comment (Silvris) [reply test/bases.py:300]: Looking at it, Location does have a `_repr_` that calls `_str_`. We could use `str(location)` instead, but the end result is the same.

**Review by NewSoupVi — APPROVED** [review]
Fast tracking this a bit because the reproducability issue is quite pressing.

If there is a better way to do the location serialisation thing, I will be happy to look at another PR for it

**Pull request merged**